cluster/addons/metrics-server/metrics-server-deployment.yaml

```diff
@@ -54,17 +54,29 @@ spec:
         command:
         - /metrics-server
         - --metric-resolution=30s
-        # These are needed for GKE, which doesn't support secure communication yet.
-        # Remove these lines for non-GKE clusters, and when GKE supports token-based auth.
-        - --kubelet-port=10255
-        - --deprecated-kubelet-completely-insecure=true
+        - --kubelet-use-node-status-port
+        - --kubelet-insecure-tls
         - --kubelet-preferred-address-types=InternalIP,Hostname,InternalDNS,ExternalDNS,ExternalIP
         - --cert-dir=/tmp
-        - --secure-port=443
+        - --secure-port=4443
```

Comment:
this changes operating characteristics in clusters with firewalls set up between master/nodes that already specifically open port 443…

Reply:
:( how can I fix this binding issue then?
https://github.com/kubernetes/kubernetes/pull/103713#issuecomment-882854865
the capabilities option doesn't make any difference and it only fails in the dockershim jobs

Reply:
> the Apiserver doesn't access the deployment pods directly, right?

it does when the apiserver is running in `--enable-aggregator-routing` mode

Reply:
I'm not familiar with the aggregator-routing mode, I mean with the metrics-server pods specifically :sweat_smile: